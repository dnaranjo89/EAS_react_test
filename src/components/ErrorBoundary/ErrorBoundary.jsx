import { Component } from 'react';
import PropTypes from 'prop-types';

class ErrorBoundary extends Component {
  constructor(props) {
    super(props);
    this.state = {
      hasError: false,
      error: null,
      errorInfo: null,
    };
  }

  componentDidCatch(error, errorInfo) {
    this.setState({ hasError: true, error, errorInfo });
    window.Raven.captureException(error, { extra: errorInfo });
  }

  render() {
    const { hasError, error, errorInfo } = this.state;
    if (hasError) {
      return this.props.render(error, errorInfo);
    }
    return this.props.children;
  }
}

ErrorBoundary.propTypes = {
  children: PropTypes.node.isRequired,
  render: PropTypes.func.isRequired,
};

export default ErrorBoundary;
