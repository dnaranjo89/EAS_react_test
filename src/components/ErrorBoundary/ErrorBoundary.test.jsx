import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ErrorBoundary from './ErrorBoundary.jsx';

const ErrorChild = () => {
  throw Error('Exception stuff');
};

const swallowErrors = codeToRun => {
  const { error } = console; // eslint-disable-line no-console
  console.error = () => {}; // eslint-disable-line no-console
  codeToRun();
  console.error = error; // eslint-disable-line no-console
};

describe('ErrorBoundary', () => {
  beforeAll(() => {
    window.Raven = { captureException: jest.fn(() => {}) };
  });
  it('Should catch errors', () => {
    swallowErrors(() => {
      const wrapper = mount(
        <ErrorBoundary render={() => 'Fire, this is fine'}>
          <ErrorChild />
        </ErrorBoundary>,
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
  it('Should try to send the error to Sentry', () => {
    swallowErrors(() => {
      shallow(
        <ErrorBoundary render={() => 'Fire, this is fine'}>
          <ErrorChild />
        </ErrorBoundary>,
      );
      expect(window.Raven.captureException).toHaveBeenCalled();
    });
  });
});
