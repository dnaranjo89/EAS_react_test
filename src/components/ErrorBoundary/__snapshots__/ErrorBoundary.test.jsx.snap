// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary Should catch errors 1`] = `
<ErrorBoundary
  render={[Function]}
>
  Fire, this is fine
</ErrorBoundary>
`;
